apps/web/pages/upgrade.tsx

```diff
@@ -0,0 +1,69 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+
+import Shell from "@calcom/features/shell/Shell";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { RouterOutputs } from "@calcom/trpc/react";
+import { trpc } from "@calcom/trpc/react";
+import { Button, EmptyScreen } from "@calcom/ui";
+import { showToast } from "@calcom/ui";
+import { ArrowUpCircle, CheckCircle } from "@calcom/ui/components/icon";
+
+import PageWrapper from "@components/PageWrapper";
+
+export type OrgUpgradeBannerProps = {
+  data: RouterOutputs["viewer"]["getUserTopBanners"]["orgUpgradeBanner"];
+};
+
+export default function UpgradePage() {
+  const { t } = useLocale();
+
+  const router = useRouter();
+  const publishOrgMutation = trpc.viewer.organizations.publish.useMutation({
+    onSuccess(data) {
+      router.push(data.url);
+    },
+    onError: (error) => {
+      showToast(error.message, "error");
+    },
+  });
+
+  /* TODO: figure out how data works 
+  if (!data) return null;
+
+  const [membership] = data;
+  if (!membership) return null; */
```

Comment:
couldnt figure this out since the OrgUpgradeBanner got some generic {...props} with data, not sure what data is.